lib/audited/auditor.rb

```diff
@@ -52,6 +52,8 @@ def audited(options = {})
 
         if audited_options[:comment_required]
           validate :presence_of_audit_comment
+          before_create :require_comment if audited_options[:on].include?(:create)
```

Comment:
so before we were trying to cram all of this comment requiring behavior just into the state of the object, and then for destroy we were using a callback. I think it make sense to use a callback for all three, and to have a more natural way of validating presence of an audit comment before any of these are called, which I will discuss below.